### PR TITLE
[bext-di] Rename `boost-di` to `bext-di`

### PR DESCRIPTION
**Pull request by kris-jusiak**
Problem:
- `boost-di` is not an official Boost library and it shouldn't use boost name.

Solution:
- Rename it to `di` instead.

### CODE REVIEW

File: ports/di/vcpkg.json [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {
-  "name": "boost-di",
+  "name": di",
```
Comment (vicroms) [inline ports/di/vcpkg.json:2]:
```suggestion
  "name": "di",
```

Missing an opening `"`.

**Review by vicroms — CHANGES_REQUESTED** [review]
We agree with distinguishing it from official Boost libraries, but just `di` is too much of a generic name.
 Also, we would want to make it so that the name doesn't match a `boost*` regex. 

What about renaming this `extboost-di` instead?

File: ports/di/vcpkg.json (hunk 1 above)
Comment (krzysztof-jusiak) [reply ports/di/vcpkg.json:2]: I see, however, it can't have boost in the name, what about `dependency-injection` then?

Comment (PhoebeHui) [reply ports/di/vcpkg.json:2]: How about bext-di?

Comment (krzysztof-jusiak) [reply ports/di/vcpkg.json:2]: I'm okay with bext-di :+1:

Comment (PhoebeHui) [reply ports/di/vcpkg.json:2]: @kreuzerkrieg, could you please update the port name to 'bext-di' and continue this PR?

Comment (PhoebeHui) [reply ports/di/vcpkg.json:2]: BTW,  you can format the vcpkg.json file by command './vcpkg format-manifest' ./ports/bext-di/vcpkg.json.

Comment (kreuzerkrieg) [reply ports/di/vcpkg.json:2]: hehe, how come my name suddenly popped up here? :)
will take a look at the evening

Comment (PhoebeHui) [reply ports/di/vcpkg.json:2]: Sorry for my mistakes!

**Comment by ahmedyarub** [comment]
There are still some references within vcpkg to boost-di. I don't think that you have renamed it everywhere.
Ex: https://github.com/microsoft/vcpkg/blob/master/versions/baseline.json#L539

**Pull request merged**